src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -182,6 +182,113 @@ public virtual void NextBytes(byte[] buffer)
+        ///   by index. This is used to populate a newly-created array.
+        /// </remarks>
+        public T[] GetItems<T>(T[] choices, int length)
+        {
+            ArgumentNullException.ThrowIfNull(choices);
+            return GetItems(new ReadOnlySpan<T>(choices), length);
+        }
+
+        /// <summary>
+        ///   Creates an array populated with items chosen at random from choices.
+        /// </summary>
+        /// <param name="choices">The items to use to populate the array.</param>
+        /// <param name="length">The length of array to return.</param>
+        /// <typeparam name="T">The type of array.</typeparam>
+        /// <returns>An array populated with random items.</returns>
+        /// <exception cref="ArgumentException"><paramref name="choices" /> is empty.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="length" /> is not zero or a positive number.
+        /// </exception>
+        /// <remarks>
+        ///   The method uses <see cref="Next(int)" /> to select items randomly from <paramref name="choices" />
+        ///   by index. This is used to populate a newly-created array.
+        /// </remarks>
+        public T[] GetItems<T>(ReadOnlySpan<T> choices, int length)
+        {
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            T[] items = new T[length];
+            GetItems<T>(choices, items.AsSpan());
+            return items;
+        }
+
+        /// <summary>
+        ///   Performs an in-place shuffle of an array.
+        /// </summary>
+        /// <param name="values">The array to shuffle.</param>
+        /// <typeparam name="T">The type of array.</typeparam>
+        /// <exception cref="ArgumentNullException"><paramref name="values" /> is <see langword="null" />.</exception>
+        /// <remarks>
+        ///   This method uses <see cref="Next(int, int)" /> to choose values for shuffling.
+        ///   This method is an O(n) operation.
+        /// </remarks>
+        public void Shuffle<T>(T[] values)
+        {
+            ArgumentNullException.ThrowIfNull(values);
+            Shuffle<T>(values.AsSpan());
+        }
+
+        /// <summary>
+        ///   Performs an in-place shuffle of a span.
+        /// </summary>
+        /// <param name="values">The span to shuffle.</param>
+        /// <typeparam name="T">The type of span.</typeparam>
+        /// <remarks>
+        ///   This method uses <see cref="Next(int, int)" /> to choose values for shuffling.
+        ///   This method is an O(n) operation.
+        /// </remarks>
+        public void Shuffle<T>(Span<T> values)
+        {
+            int n = values.Length;
+
+            for (int i = 0; i < n - 1; i++)
+            {
+                int j = Next(i, n);
```

Comment:
Same comment about measuring providing a shuffle implementation on each of our derived implementation types, rather than incurring O(N) virtual method calls.